modules/aruco/test/test_arucodetection.cpp

```diff
@@ -257,32 +280,45 @@ void CV_ArucoDetectionPerspective::run(int aprilDecimate) {
     Ptr<aruco::Dictionary> dictionary = aruco::getPredefinedDictionary(aruco::DICT_6X6_250);
 
     // detect from different positions
-    for(double distance = 0.1; distance <= 0.5; distance += 0.2) {
-        for(int pitch = 0; pitch < 360; pitch += 60) {
-            for(int yaw = 30; yaw <= 90; yaw += 50) {
+    for(double distance = 0.1; distance < 0.7; distance += 0.2) {
+        for(int pitch = 0; pitch < 360; pitch += (distance == 0.1? 60:180)) {
+            for(int yaw = 70; yaw <= 120; yaw += 40){
                 int currentId = iter % 250;
                 int markerBorder = iter % 2 + 1;
                 iter++;
                 vector< Point2f > groundTruthCorners;
-                // create synthetic image
-                Mat img =
-                    projectMarker(dictionary, currentId, cameraMatrix, deg2rad(yaw), deg2rad(pitch),
-                                  distance, imgSize, markerBorder, groundTruthCorners);
 
-                // detect markers
-                vector< vector< Point2f > > corners;
-                vector< int > ids;
                 Ptr<aruco::DetectorParameters> params = aruco::DetectorParameters::create();
                 params->minDistanceToBorder = 1;
                 params->markerBorderBits = markerBorder;
 
-                if(aprilDecimate == 1){
-                        params->cornerRefinementMethod = cv::aruco::CORNER_REFINE_APRILTAG;
+                /// create synthetic image
+                Mat img;
+                // marker :: Inverted
+                if(CV_ArucoDetectionPerspective::DETECT_INVERTED_MARKER == tryWith){
+                    img =
+                    projectMarker(dictionary, currentId, cameraMatrix, deg2rad(yaw), deg2rad(pitch),
+                                      distance, imgSize, markerBorder, groundTruthCorners, szEnclosed);
```

Comment:
`img = projectMarker()` can be moved out of "if" block.
(Left "as is" original code of `img` initialization)

```
if(CV_ArucoDetectionPerspective::DETECT_INVERTED_MARKER == tryWith)
{
    img = ~img;
    params->detectInvertedMarker = true;
}
```